beamer/tests/test_request.py

```diff
@@ -107,9 +107,8 @@ def test_challenge_own_claim(config, request_manager, token):
         ),
         int(time.time()),
     )
-
     assert not maybe_challenge(
-        claim, request_manager, to_checksum_address(agent.address)
+        claim, agent._event_processor._context
     ), "Tried to challenge own claim"
```

Comment:
I think we could make this better by 
- moving `Context` to `agent.py` 
- having the agent create context -- it is creating contracts and request tracker anyway
- provide a public property `context` on the agent

How does that sound?

Reply:
Good, will implement this.